### PR TITLE
Adopting the network service changes (in Core3.0.0)

### PR DESCRIPTION
**Pull request by yangyansong-adbe**
In Android Core 3.0.0, the network service returns a null connection when the device is offline. We are adopting this change in the Audience extension. For reference, a similar change has been made in the Identity extension (https://github.com/adobe/aepsdk-core-android/pull/617/files).

### CODE REVIEW

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 1]
```diff
@@ -710,8 +710,14 @@ private void submitSignal(final Event event) {
...
         Log.debug(LOG_TAG, LOG_SOURCE, "Queuing hit for url: %s", requestUrl);
-        AudienceDataEntity entity = new AudienceDataEntity(event, requestUrl, timeout);
-        hitQueue.queue(entity.toDataEntity());
+
+        // URLBuilder will return null, if the final URL is invalid. Drop the hit in that case.
+        if (requestUrl != null) {
+            AudienceDataEntity entity = new AudienceDataEntity(event, requestUrl, timeout);
+            hitQueue.queue(entity.toDataEntity());
+        } else {
+            Log.warning(LOG_TAG, LOG_SOURCE, "submitSignal : drop the invalid url.");
```
Comment (kevinlind) [inline code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:719]: Suggest rewording: "submitSignal - Dropping request with event id '%s' because URL is invalid", event.uniqueIdentifier

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 2]
```diff
@@ -798,6 +804,12 @@ private void sendOptOutHit(final Map<String, Object> configuration) {
                             networkRequest,
                             connection -> {
                                 if (connection == null) {
+                                    Log.trace(
+                                            LOG_TAG,
+                                            LOG_SOURCE,
+                                            "processDestsArray - Failed to send the optOut hit with"
+                                                + " connection status because the connection  is"
+                                                + " null(device is offline).");
```
Comment (kevinlind) [inline code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:812]: Reword: "sendOptOutHit - Failed to send the opt-out hit because the connection is null (network is offline)."

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 3]
```diff
@@ -1178,6 +1190,12 @@ private void processDestsArray(final JSONObject jsonResponse, final int timeout)
                                     request,
                                     connection -> {
                                         if (connection == null) {
+                                            Log.trace(
+                                                    LOG_TAG,
+                                                    LOG_SOURCE,
+                                                    "processDestsArray - Failed to process dest"
+                                                        + " because the connection  is null(device"
+                                                        + " is offline).");
```
Comment (kevinlind) [inline code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:1198]: reword: "processDestsArray - Failed to forward destinations because the connection is null (network is offline)."

**Comment by yangyansong-adbe** [comment]
@kevinlind  your review comments have been addressed. https://github.com/adobe/aepsdk-audience-android/pull/39/commits/df569ea1a098eef0a6baaf9130a49a02f8974c7f

**Review by cacheung — APPROVED** [review]

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 4]
```diff
@@ -710,8 +710,18 @@ private void submitSignal(final Event event) {
...
+
+        // URLBuilder will return null, if the final URL is invalid. Drop the hit in that case.
+        if (requestUrl != null) {
+            AudienceDataEntity entity = new AudienceDataEntity(event, requestUrl, timeout);
+            hitQueue.queue(entity.toDataEntity());
+        } else {
+            Log.warning(
+                    LOG_TAG,
+                    LOG_SOURCE,
+                    "submitSignal - Dropping request with event id '%s' because URL is invalid",
```
Comment (kevinlind) [inline code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:722]: nit: add period at the end of the log message.

**Review by kevinlind — APPROVED** [review]
Looks good. One small comment to add a period '.'.
